packages/react-google-maps-api/src/components/addons/InfoBox.tsx

```diff
@@ -48,6 +49,7 @@ interface InfoBoxState {
 }
 
 export interface InfoBoxProps {
+  children?: React.ReactNode
```

Comment:
children?: ReactNode | undefined

Reply:
But doesn't `?` already say it can be undefined?

Reply:
@Primajin Correct. So actually we wouldn't need that.

@JustFly1984 
I don't know if it is your style preference, but I can also change it to:
children: ReactNode | undefined
Or leave it as:
children?: ReactNode

Let me know!

Reply:
@JustFly1984 any comments?

Reply:
@marduzca @Primajin There is a subtle difference of not plugging the value, and plugging the value of undefined.
It is uncovered by enabling tsconfig flag "exactOptionalPropertyTypes": true

Consumer code can have both cases - then prop is not initialized, and then undefined passed to the prop.

Reply:
So should I do it like this and then we can merge it?

children?: ReactNode | undefined

Reply:
Can you please make sure that storybook and example gatsby works with react@18 too?

Reply:
@JustFly1984 
I tried both already. Storybook works perfectly and gatsby prebuild also works good and when starting it works until I reach a point in which I'm asked for a password. As I don't know the password I cannot proceed. Is that good enough?

Reply:
Sorry but which password are you talking about?
There is a field for a google api key, so you can test it locally or online.

Reply:
I probably don't know how to run the gatsby example properly and I don't find any documentation to help, but basically what I'm doing is:
```
cd packages/react-google-maps-api-gatsby-example
yarn build
yarn start
```

After running start this is what happens:
```
marduzca: react-google-maps-api-gatsby-example git:(develop) yarn start
yarn run v1.22.18
$ yarn develop
$ cross-env NODE_TLS_REJECT_UNAUTHORIZED=0 gatsby develop -S
info setting up automatic SSL certificate (may require elevated permissions/sudo)


Password:
```

Reply:
I already added my api key to the Storybook example, but with gatsby I don't find any place to add an api key

Reply:
@marduzca  This password only for generate SSL certificate to run https locally

I've made some improvements on top of your PR and created new one: https://github.com/JustFly1984/react-google-maps-api/pull/2993

Please pull my new branch and continue there. I will release new 2.9.0 version after you can test it with your google key, cos I have mine expired and do not have ability to pay for google service, due to current war in Ukraine, sanctions to Russia and me being a holder of Russian passport and my cards being useless outside of Russia.

Reply:
@marduzca This is your very own password that you set for your own computer - it just needs elevated permission (aka. admin / sudo / root) so it can generate the SSL cert locally for you.

Reply:
@marduzca you can't really use google cloud api without https

Reply:
@JustFly1984 @Primajin 
Thanks for the help. I can now confirm that gatsby works perfectly as well.  I don't need to add any more changes, so feel free to just continue with that branch on your own.

Reply:
I'm rather a fly on the wall here, as Google maps is the last bit I need to be able to upgrade to React 18 😅 - but happy if I can help with anything